llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2642,8 +2642,8 @@ namespace {
 
 struct CSEDenseMapInfo {
   static bool canHandle(const Instruction *I) {
-    return isa<InsertElementInst>(I) || isa<ExtractElementInst>(I) ||
-           isa<ShuffleVectorInst>(I) || isa<GetElementPtrInst>(I);
+    return isa<InsertElementInst, ExtractElementInst, ShuffleVectorInst,
+               GetElementPtrInst, BitCastInst>(I);
```

Comment:
I did a blame and found that this is really old code, that should probably be moved into VPlanTransforms: if you like, you can try to work on this as a follow-up?